components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -2162,6 +2181,66 @@ public ApplicationBasicInfo[] getApplicationBasicInfo(String filter, int offset,
         return appInfo.toArray(new ApplicationBasicInfo[0]);
     }
 
+    /**
+     * Get query to exclude system portals if excludeSystemPortals is true.
+     *
+     * @param excludeSystemPortals  Exclude system portals.
+     * @param useTableNameInQuery   Use the table name in query.
+     * @return String Query to exclude system portals.
+     */
+    private String populateSystemPortalsExcludeQuery(Boolean excludeSystemPortals, Boolean useTableNameInQuery) {
+
+        if (excludeSystemPortals) {
+            String excludeSystemPortalsQuery = useTableNameInQuery ?
+                    ApplicationMgtDBQueries.EXCLUDE_SYSTEM_PORTALS_BY_TABLE_NAME_AND_NAME :
+                    ApplicationMgtDBQueries.EXCLUDE_SYSTEM_PORTALS_BY_NAME;
+            Set<String> systemPortals = getSystemPortalsFromConfiguration(SYSTEM_PORTALS);
+            return systemPortals.isEmpty() ? "" :
+                    String.format(excludeSystemPortalsQuery, systemPortals.stream()
+                                    .map(s -> "'" + s + "'")
+                                    .collect(Collectors.joining(", ")));
+        } else {
+            return "";
+        }
+    }
+
+    /**
+     * Get system portals from configuration.
+     *
+     * @param parentElement Configuration tag name.
+     * @return Set of system portals from default.json.
+     */
+    private static Set<String> getSystemPortalsFromConfiguration(String parentElement) {
```

Comment:
Should be able to use IdentityUtil.getPropertyAsList instead

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/6026/commits/9ab5dadbb9c0b0f8ceffa6a997e49f282954572b